src/components/v2/Layout/Header/index.tsx

```diff
@@ -19,19 +24,35 @@ const Header = () => {
   const isOnMobile = useIsMdDown();
 
   const title = useMemo(() => {
+    // Handle special case of Market Details page
```

Comment:
If we write this as a component not in the render scope we wouldn't need to memoize it.